### PR TITLE
202410 updates

### PR DESCRIPTION
**Pull request by Chaffelson**
This pull request includes various changes to the `nipyapi` codebase, focusing on code simplification, addition of new functionalities, and removal of outdated comments. The most important changes include the addition of key password support, simplification of attribute setting, and the removal of outdated TODO comments.

### Code Simplification:
* Simplified the use of `setattr` in `recurse_flow`, `flatten`, and `list_all_by_kind` methods in `nipyapi/canvas.py`. [[1]](diffhunk://#diff-00c4fe8b57a00bcd1d9f9c5c5a5faf5e08a52ae460812ea84fa815c2d2dc9306L70-R70) [[2]](diffhunk://#diff-00c4fe8b57a00bcd1d9f9c5c5a5faf5e08a52ae460812ea84fa815c2d2dc9306L185-R182) [[3]](diffhunk://#diff-00c4fe8b57a00bcd1d9f9c5c5a5faf5e08a52ae460812ea84fa815c2d2dc9306L195-R191) [[4]](diffhunk://#diff-00c4fe8b57a00bcd1d9f9c5c5a5faf5e08a52ae460812ea84fa815c2d2dc9306L1371-R1366)

### New Functionalities:
* Added support for `key_password` in the `Configuration` class and its usage in `nipyapi/nifi/rest.py` and `nipyapi/registry/rest.py`. [[1]](diffhunk://#diff-f39bf716ddad492c0daf08b4b85ce184dbce425523bf2db69ad42be20fb0a11eR89-R90) [[2]](diffhunk://#diff-77c49e0f06c9f9b20414882539c3ee925c6d924d2f3001eb062410f33f2754c7R88-R90) [[3]](diffhunk://#diff-77c49e0f06c9f9b20414882539c3ee925c6d924d2f3001eb062410f33f2754c7R106) [[4]](diffhunk://#diff-77c49e0f06c9f9b20414882539c3ee925c6d924d2f3001eb062410f33f2754c7R118) [[5]](diffhunk://#diff-77c49e0f06c9f9b20414882539c3ee925c6d924d2f3001eb062410f33f2754c7R130) [[6]](diffhunk://#diff-8343d9932f1d5ef43bae17c6ec996b1c3672540fa0de206da66bdb6d047071b1R89-R90) [[7]](diffhunk://#diff-ca352a059f05d4b230d42b12df02396fbe675955a88c21a13cd3bdc56e18657dR88-R90) [[8]](diffhunk://#diff-ca352a059f05d4b230d42b12df02396fbe675955a88c21a13cd3bdc56e18657dR106) [[9]](diffhunk://#diff-ca352a059f05d4b230d42b12df02396fbe675955a88c21a13cd3bdc56e18657dR118) [[10]](diffhunk://#diff-ca352a059f05d4b230d42b12df02396fbe675955a88c21a13cd3bdc56e18657dR130)

### Removal of Outdated Comments:
* Removed outdated TODO comments in various files including `nipyapi/canvas.py`, `nipyapi/demo/fdlc.py`, and `nipyapi/demo/secure_connection.py`. [[1]](diffhunk://#diff-00c4fe8b57a00bcd1d9f9c5c5a5faf5e08a52ae460812ea84fa815c2d2dc9306L998) [[2]](diffhunk://#diff-72b953928b597ea26e4c04606adefd4246ad0d019790bb3346dbb28520498281L449) [[3]](diffhunk://#diff-ee2cab5f5971dffe93ea520275f547b3b6728a5eda18d4b03aa4d6ba4323f6e9L94)

### Code Formatting:
* Improved code readability by reformatting long lines in `nipyapi/security.py`. [[1]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L74-R75) [[2]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L133-R142) [[3]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L162-R184) [[4]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L200-R210) [[5]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L217-R228) [[6]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L242-R263) [[7]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L302-R316) [[8]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L426-R442) [[9]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L453-R470) [[10]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L462-R480) [[11]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L477-R496) [[12]](diffhunk://#diff-56d050d4ba0d2161331e75cd6931aa0f7cbeeb20e127f300ead42397d0ecfa14L493-R513)

### Bug Fixes:
* Fixed the method to retrieve HTTP headers in `nipyapi/nifi/rest.py` and `nipyapi/registry/rest.py`. [[1]](diffhunk://#diff-77c49e0f06c9f9b20414882539c3ee925c6d924d2f3001eb062410f33f2754c7L309-R315) [[2]](diffhunk://#diff-ca352a059f05d4b230d42b12df02396fbe675955a88c21a13cd3bdc56e18657dL309-R315)
* Updated pylintrc to match more modern python standards

### CODE REVIEW

File: nipyapi/nifi/rest.py [hunk 1]
```diff
@@ -85,6 +85,9 @@ def __init__(self, pools_size=4, maxsize=4):
         # key file
```
Comment (ottobackwards) [inline nipyapi/nifi/rest.py:85]: why call Configuration() again and again?  Maybe we should just call it once

File: nipyapi/registry/rest.py [hunk 2]
```diff
@@ -85,6 +85,9 @@ def __init__(self, pools_size=4, maxsize=4):
         # key file
         key_file = Configuration().key_file
```
Comment (ottobackwards) [inline nipyapi/registry/rest.py:87]: same comment about calling Configuration() over and over

File: nipyapi/utils.py [hunk 3]
```diff
@@ -57,17 +57,19 @@ def dump(obj, mode='json'):
             obj=prepared_obj,
             sort_keys=True,
             indent=4
-            # default=_json_default
         )
     except TypeError as e:
         raise e
     if mode == 'json':
         return out
     if mode == 'yaml':
```
Comment (ottobackwards) [inline nipyapi/utils.py:65]: what does `safe` imply?  if it makes guarantees about behavior, maybe add a comment about that to go with the other comments?

**Comment by Chaffelson** [comment]
I was going to break this out into separate PRs @ottobackwards but the fixes are quite small.
I'll do a separate one to bring the swagger version up to date.

**Review by ottobackwards — APPROVED** [review]
lgtm

**Comment by ottobackwards** [comment]
I'm not sure how big a deal the pyup.io/safety-ci failure is or if that is pre-existing?

**Comment by Chaffelson** [comment]
> I'm not sure how big a deal the pyup.io/safety-ci failure is or if that is pre-existing?

it's a very old and irrelevant discredited vuln in pytest. I see there is a newer version of pytest that avoids it, I'll see if we can upgrade version painlessly.

**Pull request merged**